sys/usb/usbus/usbus_control.c

```diff
@@ -0,0 +1,417 @@
+        for (usbus_endpoint_t *ep = iface->ep; ep; ep = ep->next) {
+            if (ep->active) {
+                static const usbopt_enable_t enable = USBOPT_ENABLE;
+                usbdev_ep_set(ep->ep, USBOPT_EP_ENABLE, &enable,
+                              sizeof(usbopt_enable_t));
+                DEBUG("usbus_control: activated endpoint %d, dir %s\n",
+                      ep->ep->num,
+                      ep->ep->dir == USB_EP_DIR_OUT ? "out" : "in");
+            }
+        }
+        for (usbus_interface_alt_t *alt = iface->alts; alt; alt = alt->next) {
+            for (usbus_endpoint_t *ep = alt->ep; ep; ep = ep->next) {
+                if (ep->active) {
+                    static const usbopt_enable_t enable = USBOPT_ENABLE;
+                    usbdev_ep_set(ep->ep, USBOPT_EP_ENABLE, &enable,
+                                  sizeof(usbopt_enable_t));
+                    DEBUG("usbus_control: activated endpoint %d, dir %s\n",
+                          ep->ep->num,
+                          ep->ep->dir == USB_EP_DIR_OUT ? "out" : "in");
+                }
+            }
+        }
+    }
+}
+
+static size_t _cpy_str_to_utf16(usbus_t *usbus, const char *str)
+{
+    size_t len = 0;
+
+    while (*str) {
+        usbus_control_slicer_put_char(usbus, *str);
+        usbus_control_slicer_put_char(usbus, 0);
+        len += 2; /* Two bytes added each iteration */
+        str++;
+    }
+    return len;
+}
+
+static usbus_string_t *_get_descriptor(usbus_t *usbus, uint16_t idx)
+{
+    for (usbus_string_t *str = usbus->strings; str; str = str->next) {
+        if (str->idx == idx) {
+            return str;
+        }
+    }
+    return NULL;
+}
+
+static int _req_status(usbus_t *usbus)
+{
+    uint8_t status[2];
+
+    memset(status, 0, sizeof(status));
+    usbus_control_slicer_put_bytes(usbus, status, sizeof(status));
+    return sizeof(status);
+}
+
+static int _req_str(usbus_t *usbus, uint16_t idx)
+{
+    if (idx == 0) {
+        usb_descriptor_string_t desc;
+        desc.type = USB_TYPE_DESCRIPTOR_STRING;
+        desc.length = sizeof(uint16_t) + sizeof(usb_descriptor_string_t);
+        usbus_control_slicer_put_bytes(usbus, (uint8_t *)&desc, sizeof(desc));
+        /* Only one language ID supported */
+        uint16_t us = USB_CONFIG_DEFAULT_LANGID;
+        usbus_control_slicer_put_bytes(usbus, (uint8_t *)&us, sizeof(uint16_t));
+    }
+    else {
+        usb_descriptor_string_t desc;
+        desc.type = USB_TYPE_DESCRIPTOR_STRING;
+        usbus_string_t *str = _get_descriptor(usbus, idx);
+        if (str) {
+            desc.length = sizeof(usb_descriptor_string_t);
+            desc.length += 2 * strlen(str->str); /* USB strings are UTF-16 */
+            usbus_control_slicer_put_bytes(usbus, (uint8_t *)&desc,
+                                           sizeof(desc));
+            _cpy_str_to_utf16(usbus, str->str);
+        }
+    }
+    return 1;
```

Comment:
Is it needed to return 1 all the time ?

Reply:
No, I should fix the function to return something different when the requested string is not found :/

Reply:
Fixed function to also be able to  return `-1` in some cases